packages/connect-node/src/node-universal-client.ts

```diff
@@ -317,8 +317,28 @@ function h2Request(
         }
         sentinel.reject(e);
       });
-
+      let receivedData = false;
+      stream.on("response", () => {
+        receivedData = true;
+      });
```

Comment:
It's sensible to raise an error for a RST_STREAM with NO_ERROR received before response headers. But I don't think it fully addresses https://github.com/connectrpc/connect-es/issues/1108. The issue mentions an open streaming RPC, so the situation is much more likely to occur _after_ response headers have already been received. 

Ideally, the logic would be to raise an error for any RST_STREAM with NO_ERROR, unless we already received a status in headers or trailers.

This is impossible to implement in the universal client, since it doesn't know the status (which could be encoded in the body, or in trailers, depending on the protocol). 

But can't we verify that we receive a trailer with a grpc-status field here? https://github.com/connectrpc/connect-es/blob/v1.4.0/packages/connect/src/protocol-grpc/transport.ts#L248 It looks like we currently only raise an error from the function `validateTrailer` if we receive an error code, not if we're missing the field.

This will be more complex to test (we should also check unary, gRPC-web, and Connect), but I think this should address the issue exhaustively.

Reply:
Yes, definitely. And we should add conformance test cases to verify this since it's possible that we have similar issues in other client implementations.

Looking at the [specs for error codes](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md), I think this would fall into the category of "Error parsing returned status", which would map to an UNKNOWN code.

Reply:
grpcurl (as stated in the issue by the user) reports a code internal, grpc-js also uses internal (https://github.com/grpc/grpc-node/issues/2569). I wonder what go does.